<a href="https://colab.research.google.com/github/CasillasAlba/GeneFlow/blob/main/colab_files/Prueba_Workflow_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ============================================================
# CONEXIÓN CON GOOGLE DRIVE
# ============================================================

In [1]:
from google.colab import drive
drive.mount("/content/drive") # Se monta el Drive

from google.colab import files

Mounted at /content/drive


# ============================================================
# USO DE FICHEROS .PY COMO MÓDULO EN COLAB
# ============================================================

In [2]:
!cp /content/drive/MyDrive/TFG/GeneFlow/src/dataobject.py /content
!cp /content/drive/MyDrive/TFG/GeneFlow/src/etl.py /content
!cp /content/drive/MyDrive/TFG/GeneFlow/src/model.py /content
!cp /content/drive/MyDrive/TFG/GeneFlow/src/objects.py /content
!cp /content/drive/MyDrive/TFG/GeneFlow/src/processing.py /content
!cp /content/drive/MyDrive/TFG/GeneFlow/src/utils.py /content
!cp /content/drive/MyDrive/TFG/GeneFlow/src/visualize.py /content

!cp /content/drive/MyDrive/TFG/GeneFlow/geneflow.py /content

In [3]:
!jupyter notebook --NotebookApp.iopub_data_rate_limit=1.0e10

Traceback (most recent call last):
  File "/usr/local/bin/jupyter-notebook", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/dist-packages/jupyter_core/application.py", line 269, in launch_instance
    return super().launch_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/traitlets/config/application.py", line 845, in launch_instance
    app.initialize(argv)
  File "/usr/local/lib/python3.7/dist-packages/traitlets/config/application.py", line 88, in inner
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/notebook/notebookapp.py", line 1507, in initialize
    self.init_webapp()
  File "/usr/local/lib/python3.7/dist-packages/notebook/notebookapp.py", line 1297, in init_webapp
    self.http_server.listen(port, self.ip)
  File "/usr/local/lib/python3.7/dist-packages/tornado/tcpserver.py", line 143, in listen
    sockets = bind_sockets(port, address=address)
  File "/usr/local/lib/python3.7/dist-packa

# **REFLEXION CON DATA OBJECT**


# ============================================================
# IMPORTAMOS GENEFLOW
# ============================================================

In [4]:
import geneflow as genf

# ============================================================
# LECTURA DE DATOS Y CREACIÓN DE UN OBJETO DATAOBJECT
# ============================================================

In [18]:
datos = genf.read_file("/content/drive/MyDrive/TFG/GeneFlow/data/countdata.txt", index = "Gene_ID")

obs_info = genf.read_file("/content/drive/MyDrive/TFG/GeneFlow/data/SampleInfo_Corrected.txt", index = "SampleName")

obs_info = obs_info.iloc[:, 1:]

data_object = genf.create_data_object(datos, obs_ = obs_info)

data_object.summary_object()

print("\n")

(data_object.get_counts()).head()


Creating DataObject...


##############################################

Data Object
	Dimensions:  (27179, 12)
	Row Names (27179): 497097 ... 100504472
	Column Names (12): MCL1.DG ... MCL1.LF

	Var
		There is no data for Var

	Obs
		Dimensions:  (12, 2)
		Row Names (12): MCL1.DG ... MCL1.LF
		Column Names (2): CellType ... Status

##############################################




MCL1.DG  MCL1.DH  MCL1.DI  MCL1.DJ  MCL1.DK  MCL1.DL  MCL1.LA  \
Gene_ID                                                                    
497097       438.0    300.0     65.0    237.0    354.0    287.0      0.0   
100503874      1.0      0.0      1.0      1.0      0.0      4.0      0.0   
100038431      0.0      0.0      0.0      0.0      0.0      0.0      0.0   
19888          1.0      1.0      0.0      0.0      0.0      0.0     10.0   
20671        106.0    182.0     82.0    105.0     43.0     82.0     16.0   

           MCL1.LB  MCL1.LC  MCL1.LD  MCL1.LE  MCL1.LF  
Gene_ID                                                 
497097         0.0      0.0      0.0      0.0      0.0  
100503874      0.0      0.0      0.0      0.0      0.0  
100038431      0.0      0.0      0.0      0.0      0.0  
19888          3.0     10.0      2.0      0.0      0.0  
20671         25.0     18.0      8.0      3.0     10.0

Se crea una copia de los datos en bruto para replicando el flujo de trabajo aplicando la reflexión.

In [6]:
data_object_copy = genf.copy_object(data_object)

# ============================================================
# NORMALIZACIÓN DE DATOS CON CPM
# ============================================================

Se realiza la normalización de los datos usando CPM (counts per million).

In [8]:
print("\nCOUNT CPM")
data_object = genf.counts_cpm(data_object)

(data_object.get_counts()).head()


COUNT CPM


MCL1.DG    MCL1.DH   MCL1.DI    MCL1.DJ    MCL1.DK    MCL1.DL  \
Gene_ID                                                                      
497097     18.856844  13.775439  2.697010  10.456480  16.442685  14.338969   
100503874   0.043052   0.000000  0.041492   0.044120   0.000000   0.199846   
100038431   0.000000   0.000000  0.000000   0.000000   0.000000   0.000000   
19888       0.043052   0.045918  0.000000   0.000000   0.000000   0.000000   
20671       4.563528   8.357099  3.402382   4.632618   1.997275   4.096848   

            MCL1.LA   MCL1.LB   MCL1.LC   MCL1.LD  MCL1.LE  MCL1.LF  
Gene_ID                                                              
497097     0.000000  0.000000  0.000000  0.000000  0.00000  0.00000  
100503874  0.000000  0.000000  0.000000  0.000000  0.00000  0.00000  
100038431  0.000000  0.000000  0.000000  0.000000  0.00000  0.00000  
19888      0.490386  0.138197  0.449608  0.090958  0.00000  0.00000  
20671      0.784617  1.151641  0.809294  0.363831  0.12134  0.40556

# ============================================================
# ELIMINACIÓN DE GENES CON BAJO NÚMERO DE LECTURAS
# ============================================================

Se eliminan los genes con un bajo número de lecturas por muestra. Como se puede ver, se ha pasado de tener 27179 genes (filas de la matriz de conteo) a 15804.

In [9]:
threshs = genf.remove_low_reads(data_object.get_counts(), thresh = 0.5)

subdf = genf.data_selection_filter(data_object.get_counts(), filter_= threshs)

proc_df = genf.drop_values(subdf, axis = 1) 

genf.set_counts(data_object, proc_df)

print("Resultado después de eliminar genes poco expresados\n")
data_object.summary_object()
(data_object.get_counts()).head()

Resultado después de eliminar genes poco expresados


##############################################

Data Object
	Dimensions:  (15804, 12)
	Row Names (15804): 497097 ... 170942
	Column Names (12): MCL1.DG ... MCL1.LF

	Var
		There is no data for Var

	Obs
		Dimensions:  (12, 2)
		Row Names (12): MCL1.DG ... MCL1.LF
		Column Names (2): CellType ... Status

##############################################


MCL1.DG    MCL1.DH    MCL1.DI    MCL1.DJ    MCL1.DK    MCL1.DL  \
Gene_ID                                                                     
497097   18.856844  13.775439   2.697010  10.456480  16.442685  14.338969   
20671     4.563528   8.357099   3.402382   4.632618   1.997275   4.096848   
27395    13.303116  10.744842  13.982959  13.236051  13.469996  13.489622   
18777    28.070005  23.647836  39.334851  41.252358  43.103987  39.519598   
21399    69.055657  68.647602  71.408522  58.106263  53.833535  53.259028   

           MCL1.LA    MCL1.LB    MCL1.LC    MCL1.LD    MCL1.LE    MCL1.LF  
Gene_ID                                                                    
497097    0.000000   0.000000   0.000000   0.000000   0.000000   0.000000  
20671     0.784617   1.151641   0.809294   0.363831   0.121340   0.405560  
27395    27.461598  21.374459  21.985821  14.917065  12.417172  13.870136  
18777    40.505856  39.708585  30.033801  29.379341  22.003066  23.563008  
21399    65.417448  57.950580  48.018113  42.113421  20.546981  20.277976

# ============================================================
# SE GUARDA EN UN FICHERO EL FLUJO DE TRABAJO
# ============================================================

In [13]:
genf.workflow_to_json(data_object.get_workflow(), path = "/content/drive/MyDrive/TFG/GeneFlow/data/Reflection/workflow.json")

# ============================================================
# DATOS EN BRUTO SOBRE LOS QUE APLICAR EL FLUJO DE TRABAJO
# ============================================================

In [15]:
data_object_copy.summary_object()

print("\n")

(data_object_copy.get_counts()).head()


##############################################

Data Object
	Dimensions:  (27179, 12)
	Row Names (27179): 497097 ... 100504472
	Column Names (12): MCL1.DG ... MCL1.LF

	Var
		There is no data for Var

	Obs
		Dimensions:  (12, 2)
		Row Names (12): MCL1.DG ... MCL1.LF
		Column Names (2): CellType ... Status

##############################################




MCL1.DG  MCL1.DH  MCL1.DI  MCL1.DJ  MCL1.DK  MCL1.DL  MCL1.LA  \
Gene_ID                                                                    
497097       438.0    300.0     65.0    237.0    354.0    287.0      0.0   
100503874      1.0      0.0      1.0      1.0      0.0      4.0      0.0   
100038431      0.0      0.0      0.0      0.0      0.0      0.0      0.0   
19888          1.0      1.0      0.0      0.0      0.0      0.0     10.0   
20671        106.0    182.0     82.0    105.0     43.0     82.0     16.0   

           MCL1.LB  MCL1.LC  MCL1.LD  MCL1.LE  MCL1.LF  
Gene_ID                                                 
497097         0.0      0.0      0.0      0.0      0.0  
100503874      0.0      0.0      0.0      0.0      0.0  
100038431      0.0      0.0      0.0      0.0      0.0  
19888          3.0     10.0      2.0      0.0      0.0  
20671         25.0     18.0      8.0      3.0     10.0

# ============================================================
# SERIALIZACIÓN DEL FICHERO JSON Y REPRODUCCIÓN DEL FLUJO DE TRABAJO
# ============================================================

In [16]:
print("Using reflection (reconstructing objects from dictionaries)...")

json_workflow = genf.workflow_from_json(path = "/content/drive/MyDrive/TFG/GeneFlow/data/Reflection/workflow.json")

new_data_object = genf.replicate_workflow(json_workflow, data_object_copy)

Using reflection (reconstructing objects from dictionaries)...


# ============================================================
# RESULTADO TRAS APLICAR REFLEXIÓN
# ============================================================

Como se puede comprobar, el DataObject obtenido es el mismo tras replicar el flujo de trabajo.

In [17]:
print("Using reflection (reconstructing objects from dictionaries)...")

new_data_object.summary_object()

print("\n")

(new_data_object.get_counts()).head()

Using reflection (reconstructing objects from dictionaries)...

##############################################

Data Object
	Dimensions:  (15804, 12)
	Row Names (15804): 497097 ... 170942
	Column Names (12): MCL1.DG ... MCL1.LF

	Var
		There is no data for Var

	Obs
		Dimensions:  (12, 2)
		Row Names (12): MCL1.DG ... MCL1.LF
		Column Names (2): CellType ... Status

##############################################




MCL1.DG    MCL1.DH    MCL1.DI    MCL1.DJ    MCL1.DK    MCL1.DL  \
Gene_ID                                                                     
497097   18.856844  13.775439   2.697010  10.456480  16.442685  14.338969   
20671     4.563528   8.357099   3.402382   4.632618   1.997275   4.096848   
27395    13.303116  10.744842  13.982959  13.236051  13.469996  13.489622   
18777    28.070005  23.647836  39.334851  41.252358  43.103987  39.519598   
21399    69.055657  68.647602  71.408522  58.106263  53.833535  53.259028   

           MCL1.LA    MCL1.LB    MCL1.LC    MCL1.LD    MCL1.LE    MCL1.LF  
Gene_ID                                                                    
497097    0.000000   0.000000   0.000000   0.000000   0.000000   0.000000  
20671     0.784617   1.151641   0.809294   0.363831   0.121340   0.405560  
27395    27.461598  21.374459  21.985821  14.917065  12.417172  13.870136  
18777    40.505856  39.708585  30.033801  29.379341  22.003066  23.563008  
21399    65.417448  57.950580  48.018113  42.113421  20.546981  20.277976